src/pages/api/thermostats/update.ts

```diff
@@ -0,0 +1,64 @@
+import { NotFoundException } from "nextlove"
+import { z } from "zod"
+
+import { THERMOSTAT_DEVICE_TYPES } from "lib/zod/index.ts"
+
+import { withRouteSpec } from "lib/middleware/with-route-spec.ts"
+import {
+  normalizeClimateSetting,
+  refineClimateSetting,
+  throwIfClimateSettingNotAllowed,
+} from "lib/util/thermostats.ts"
+import { climate_setting } from "lib/zod/climate_setting.ts"
+
+export default withRouteSpec({
+  auth: "cst_ak_pk",
+  methods: ["PATCH", "POST"],
+  jsonBody: z.object({
+    device_id: z.string(),
+    default_climate_setting: refineClimateSetting(climate_setting.partial()),
+  }),
+  jsonResponse: z.object({}),
+} as const)(async (req, res) => {
+  const { device_id, default_climate_setting } = req.body
+
+  const device = req.db.devices.find((device) => {
+    if (!THERMOSTAT_DEVICE_TYPES.includes(device.device_type)) {
+      return false
+    }
+
+    return device.device_id === device_id
+  })
+
+  if (device == null) {
+    throw new NotFoundException({
+      type: "device_not_found",
+      message: `Could not find a thermostat with device_id`,
+      data: { device_id },
+    })
+  }
+
+  const thermostat_original_climate_setting =
+    "default_climate_setting" in device.properties
+      ? device.properties.default_climate_setting
+      : null
+
```

Comment:
Should we add this check here?
```
if (
    !thermostat_original_climate_setting &&
    !default_climate_setting["manual_override_allowed"]
  )
    throw new Error("manual_override_allowed must be defined")
```

We check for this here: https://github.com/seamapi/seam-connect/blob/main/pages/api/public/thermostats/update.ts#L37C1-L42C1

Reply:
thanks, updated in https://github.com/seamapi/fake-seam-connect/pull/101/commits/b90b964990302bb866aaeb9b2a4f23128c3a8b1c